robolectric-resources/src/main/java/org/robolectric/res/ResourceMerger.java

```diff
@@ -0,0 +1,44 @@
+package org.robolectric.res;
+
+import org.jetbrains.annotations.NotNull;
+import org.robolectric.manifest.AndroidManifest;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class  ResourceMerger {
+  @NotNull
+  public static ResourceTable buildResourceTable(AndroidManifest appManifest) {
```

Comment:
Yeah, this is exactly what I was thinking, some kind of ResourceTableBuilder, currently there are three kinds I can see:-

1) Current - Robolectric merges resources, Robolectric parse R classes +  ResourceIdGeneration
2) Build system merges resources according to its own strategy employed for building the APK. Robolectric parses R classes + ResourceIdGeneration
3) Build system merges resources + generates arsc files. Robolectric parses arsc file for everything.
4) Any custom builds for projects out there we haven't considered.

I think I'm going to leave this for now, it will be easier to do once the ResourceTable contains the indexes too then we have a single point to load resources with their IDs.